### PR TITLE
VAULT-14734: activity log write endpoint

### PR DESCRIPTION
**Pull request by miagilepner**
This PR creates a no-op endpoint at `sys/internal/counters/activity/write` for writing activity log data when Vault is compiled with the `testonly` build tag. This endpoint accepts JSON input for the data to write to the activity log. 

The input is defined as protobuf and Go models are generated from the protobuf for simplicity's sake - there are several "oneof" options in the input, and protobuf gives a built-in way to handle those restrictions. 

Currently, the endpoint only validates the input. A later PR will update the handler to generate and write the data.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,13 +30,16 @@ default: dev
 bin: prep
 	@CGO_ENABLED=$(CGO_ENABLED) BUILD_TAGS='$(BUILD_TAGS) ui' sh -c "'$(CURDIR)/scripts/build.sh'"
 
+testonly:
```
Comment (hghaf099) [inline Makefile:33]: Should this be a make target? calling make with this target seems to be doing nothing. What do you think changing it to a variable?

File: Makefile [hunk 2]
```diff
@@ -220,7 +224,7 @@ proto: bootstrap
...
 	$(SED) -i -e 's/Id/ID/' -e 's/SPDX-License-IDentifier/SPDX-License-Identifier/' vault/request_forwarding_service.pb.go
-	$(SED) -i -e 's/Idp/IDP/' -e 's/Url/URL/' -e 's/Id/ID/' -e 's/IDentity/Identity/' -e 's/EntityId/EntityID/' -e 's/Api/API/' -e 's/Qr/QR/' -e 's/Totp/TOTP/' -e 's/Mfa/MFA/' -e 's/Pingid/PingID/' -e 's/namespaceId/namespaceID/' -e 's/Ttl/TTL/' -e 's/BoundCidrs/BoundCIDRs/' -e 's/SPDX-License-IDentifier/SPDX-License-Identifier/' helper/identity/types.pb.go helper/identity/mfa/types.pb.go helper/storagepacker/types.pb.go sdk/plugin/pb/backend.pb.go sdk/logical/identity.pb.go vault/activity/activity_log.pb.go
+	$(SED) -i -e 's/Idp/IDP/' -e 's/Url/URL/' -e 's/Id/ID/' -e 's/IDentity/Identity/' -e 's/EntityId/EntityID/' -e 's/Api/API/' -e 's/Qr/QR/' -e 's/Totp/TOTP/' -e 's/Mfa/MFA/' -e 's/Pingid/PingID/' -e 's/namespaceId/namespaceID/' -e 's/Ttl/TTL/' -e 's/BoundCidrs/BoundCIDRs/' -e 's/SPDX-License-IDentifier/SPDX-License-Identifier/' helper/identity/types.pb.go helper/identity/mfa/types.pb.go helper/storagepacker/types.pb.go sdk/plugin/pb/backend.pb.go sdk/logical/identity.pb.go vault/activity/generation/generate_data.pb.go vault/activity/activity_log.pb.go
```
Comment (hghaf099) [inline Makefile:227]: Just curious, why the new file is added here?

File: vault/activity/generation/generate_data.proto [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  int32 num_segments = 7;
+}
+
+message Segments {
+  repeated Segment segments = 1;
+}
+
+message Segment {
+  optional int32 segment_index = 1;
+  repeated Client clients = 2;
```
Comment (hghaf099) [inline vault/activity/generation/generate_data.proto:41]: Can this be of type `Clients`?

File: vault/logical_system_activity_write_testonly_test.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		{
+			name:      "empty data value fails",
+			operation: logical.CreateOperation,
+			input:     map[string]interface{}{"input": `{"write":["WRITE_PRECOMPUTED_QUERIES"],"data":[]}`},
+			wantError: logical.ErrInvalidRequest,
+		},
+		{
+			name:      "correctly formatted data succeeds",
+			operation: logical.CreateOperation,
+			input:     map[string]interface{}{"input": `{"write":["WRITE_PRECOMPUTED_QUERIES"],"data":[{"current_month":true,"all":{"clients":[{"count":5}]}}]}`},
```
Comment (hghaf099) [inline vault/logical_system_activity_write_testonly_test.go:68]: Outside of this PR, is there a tool or something to generate this data?

**Review by hghaf099 — COMMENTED** [review]
Looks good! Just a few comments/questions.

File: Makefile (hunk 1 above)
Comment (raskchanky) [reply Makefile:33]: I'm guessing it's here as a dependency target, not one to call directly, as a way of making sure the appropriate targets further down in the file get the `testonly` build tag applied.

**Review by raskchanky — COMMENTED** [review]
Looks good! I gave a 👍 to some of Hamid's comments that I agreed with.

File: vault/logical_system_activity_write_testonly_test.go (hunk 4 above)
Comment (miagilepner) [reply vault/logical_system_activity_write_testonly_test.go:68]: there will be, it's a later ticket in the epic.

File: Makefile (hunk 2 above)
Comment (miagilepner) [reply Makefile]: I didn't notice the comment above saying to just accept the protobuf generated names. Removed `generate_data.pb.go` from the list

File: vault/activity/generation/generate_data.proto (hunk 3 above)
Comment (miagilepner) [reply vault/activity/generation/generate_data.proto]: Done

**Pull request merged**